src/builder/actions/CometActionsBuilder.sol

```diff
@@ -0,0 +1,350 @@
+            assetSymbolOuts[0] = repayIntent.assetSymbol;
+            actionIntent = QuarkBuilderBase.ActionIntent({
+                actor: repayIntent.repayer,
+                amountOuts: amountOuts,
+                assetSymbolOuts: assetSymbolOuts,
+                amountIns: repayIntent.collateralAmounts,
+                assetSymbolIns: repayIntent.collateralAssetSymbols,
+                blockTimestamp: repayIntent.blockTimestamp,
+                chainId: repayIntent.chainId,
+                useQuotecall: useQuotecall,
+                bridgeEnabled: true,
+                autoWrapperEnabled: true
+            });
+        }
+
+        (IQuarkWallet.QuarkOperation[] memory quarkOperationsArray, Actions.Action[] memory actionsArray) =
+        QuarkBuilderBase.collectAssetsForAction({
+            actionIntent: actionIntent,
+            chainAccountsList: chainAccountsList,
+            payment: payment,
+            actionQuarkOperation: repayQuarkOperation,
+            action: repayAction
+        });
+
+        return BuilderResult({
+            version: VERSION,
+            actions: actionsArray,
+            quarkOperations: quarkOperationsArray,
+            paymentCurrency: payment.currency,
+            eip712Data: EIP712Helper.eip712DataForQuarkOperations(quarkOperationsArray, actionsArray)
+        });
+    }
+
+    struct CometBorrowIntent {
+        uint256 amount;
+        string assetSymbol;
+        uint256 blockTimestamp;
+        address borrower;
+        uint256 chainId;
+        uint256[] collateralAmounts;
+        string[] collateralAssetSymbols;
+        address comet;
+    }
+
+    function cometBorrow(
+        CometBorrowIntent memory borrowIntent,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) external pure returns (BuilderResult memory /* builderResult */ ) {
+        if (borrowIntent.collateralAmounts.length != borrowIntent.collateralAssetSymbols.length) {
+            revert InvalidInput();
+        }
+
+        bool useQuotecall = false; // never use Quotecall
+
+        (IQuarkWallet.QuarkOperation memory borrowQuarkOperation, Actions.Action memory borrowAction) = Actions
+            .cometBorrow(
+            Actions.CometBorrowInput({
+                chainAccountsList: chainAccountsList,
+                amount: borrowIntent.amount,
+                assetSymbol: borrowIntent.assetSymbol,
+                blockTimestamp: borrowIntent.blockTimestamp,
+                borrower: borrowIntent.borrower,
+                chainId: borrowIntent.chainId,
+                collateralAmounts: borrowIntent.collateralAmounts,
+                collateralAssetSymbols: borrowIntent.collateralAssetSymbols,
+                comet: borrowIntent.comet
+            }),
+            payment
+        );
+
+        QuarkBuilderBase.ActionIntent memory actionIntent;
+        // Note:scope to avoid stack too deep errors
```

Comment:
nit: missing space between "Note:" and "scope". Same for all other places where this note is left